### ISSUE TITLE
window scrolls when dropdown is clicked

### ISSUE THREAD

**Issue by cedrictools** [Day8/re-com#5]
On re-com demo, when adding a big div before a dropdown, like the one below, clicking on the dropdown will cause some unexpected scrolling.

```diff
diff --git a/src/re_demo/dropdowns.cljs b/src/re_demo/dropdowns.cljs
index 3e7fa3e..182a39a 100644
--- a/src/re_demo/dropdowns.cljs
+++ b/src/re_demo/dropdowns.cljs
@@ -327,6 +327,7 @@
                                :width     "700px"
                                :gap       "10px"
                                :children  [[title2 "Demo"]
+                                           [:div {:style {:height "1000px"  } }]
                                            [h-box
                                             :gap      "10px"
                                             :align    :center
```

Tested with browser chromium, Version 41.0.2272.118 (64-bit)

**Comment by ducky427** [comment]
I have also encountered this issue.

**Comment by mike-thompson-day8** [comment]
Analysis: 
   - putting that large vertical height in causes a vertical scrollbar to appear on the "page"  (scrollbar appears far right)
   - when a dropdown has lots of options, it has its own scrollbar, allowing you to scroll through all the options and presumably click on one. 
   - when you click on a dropdown component, and it pops-up the options list, it internally uses [scrollIntoView](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView)  to get the currently selected item into view (this magically moves the dropdown component's scrollbar to show the currently selected option).  
   - trouble is ... that call to `scrollIntoView` ALSO causes the page's scrollbar to move AS WELL. Both scrollbars move (component and page).  And that's why  stuff suddenly starts jumping about. That's also why this only shows up when you add in the large vertical space (because that triggers a page scrollbar)

We're looking into solutions.

**Comment by mike-thompson-day8** [comment]
@cedrictools   as an aside, beware using `[:div ...]` as children to `h-box` and `v-box`. It WILL bite you sooner or later. (Unless you add all the right flex styling).

See the warning "Be All In":
http://re-demo.s3-website-ap-southeast-2.amazonaws.com/#/layout

The alternative to your `[:div ...]` would be to use:
  `[gap  :size   "1000px"]`
http://re-demo.s3-website-ap-southeast-2.amazonaws.com/#/gap

But, thanks for the bug report!

**Comment by ducky427** [comment]
@mike-thompson-day8, thanks for the comments.

I can reproduce this issue with `gap` as well. Below is the code for the whole page.

```clojure
(ns recomselect.core
    (:require [reagent.core :as reagent :refer [atom]]
              [re-com.core :as rcc]
              [cljsjs.react :as react]))


(defn home-page []
  (let [model (atom "Hello")]
    [rcc/v-box
     :height "100%"
     :children [[rcc/gap :size "1000px"]
                [rcc/single-dropdown
                 :model model
                 :width "200px"
                 :on-change #(reset! model %)
                 :choices [{:id 1 :label "Hello"}
                          {:id 2 :label "World"}
                          {:id 3 :label "Test"}]]]]))

(defn mount-root []
  (reagent/render [home-page] (.getElementById js/document "app")))

(defn init! []
  (mount-root))
```

**Comment by mike-thompson-day8** [comment]
It looks like Gregg has figured out how to fix this.  We're expecting a v0.5.2 release tomorrow.

**Comment by Gregg8** [comment]
Just released 0.5.2 to address this.
Available on Clojars
Demo updated: http://re-demo.s3-website-ap-southeast-2.amazonaws.com/#/dropdown

**Comment by ducky427** [comment]
thats great guys, @Gregg8, @mike-thompson-day8.

Thanks a lot!!

**Comment by cedrictools** [comment]
Indeed it's fixed with 0.5.2, great lib, thanks for sharing all that stuff !
